packages/lesswrong/lib/collections/posts/schema.tsx

```diff
@@ -287,6 +287,12 @@ const schema: SchemaType<DbPost> = {
     canRead: ['guests'],
     canUpdate: ['members'],
     hidden: true,
+    onUpdate: ({data, document, oldDocument, currentUser}) => {
+      if (!currentUser?.isAdmin && oldDocument.deletedDraft && !document.deletedDraft) {
```

Comment:
Good job catching this issue, this seems like a fine solution given that there isn't a way in the UI to un-delete a post